### PR TITLE
Convert unbatched inputs to batch with BS=1, and Make MHA symbolically traceable

### PR DESCRIPTION
**Pull request by mikekgfb**
Summary: Make MHA symbolically traceable

Test Plan: sandcastle

Differential Revision: D39790429

### CODE REVIEW

**Comment by mikekgfb** [comment]
________________________________

File: torch/nn/modules/activation.py [hunk 1]
```diff
@@ -1061,7 +1061,19 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
...
+        is_batched = True
+        qsize = query.size()
+        ksize = key.size()
+        vsize = value.size()
+        if self.batch_first:
+            query = query.view(-1, qsize[-2], qsize[-1])
+            key = key.view(-1, ksize[-2], ksize[-1])
+            value = value.view(-1, vsize[-2], vsize[-1])
+        else:
+            query = query.view(qsize[0], qsize[1], -1)
```
Comment (ngimel) [inline torch/nn/modules/activation.py:1073]: this doesn't look right, for non-batch-first sizes are (S,N,E) or (S,E), so it should be `.view(size(0), -1, size(-1)`

**Pull request closed without merging**